application/src/main/java/org/togetherjava/tjbot/commands/moderation/WarnCommand.java

```diff
@@ -0,0 +1,165 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.utils.Result;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.WarnSystem;
+import org.togetherjava.tjbot.db.generated.tables.records.WarnSystemRecord;
+
+import java.awt.*;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * This command allows a moderator to warn a user. This moderator will need to have ban permission
+ * and well need to type {@code /warn @user reason}. If the command is successful the user will be
+ * warned and will receive a dm telling them why there were warned. After that all the data will be
+ * added to the database.
```

Comment:
Your javadoc documents the current implementation instead of the outside looking blackbox-ish behavior. Take a look at the other javadocs for a good reference and adjust it accordingly.